### DISCUSSION TITLE
java17下搭建rocketMQ，broker_default.log 报instrumneted a sprcial java.util.Set into : sun.nio.ch.EPollSelectorImpl

### DISCUSSION THREAD

**Discussion by alloyinfo** [apache/rocketmq#7274] [Q&A]
java17下搭建rocketMQ，broker_default.log 报instrumneted a sprcial java.util.Set into : sun.nio.ch.EPollSelectorImpl

**Comment by alloyinfo** [comment]
最新版本5.1.3

**Comment by joeCarf** [comment]
RcoketMQ目前默认使用1.8进行搭建。java17存在一些封闭类，可能会导致运行时报错。

**Reply by alloyinfo** [reply]
local模式搭建成功了，集群没有成功

**Reply by alloyinfo** [reply]
java17下的rocketmq就算不报包的问题，过了几十分钟就会proxy就自己挂掉了

**Reply by joeCarf** [reply]
目前兼容性应该有问题，建议还是使用1.8进行搭建

**Reply by alloyinfo** [reply]
换了java8也报这个错

**Reply by alloyinfo** [reply]
在broker_default.log中打印的这个错误，jps有namesrv和proxy

**Comment by mxsm** [comment]
> java17下搭建rocketMQ，broker_default.log 报instrumneted a sprcial java.util.Set into : sun.nio.ch.EPollSelectorImpl

现在Rocketmq的版本还是在Jdk8, JDK9+会出现一些问题。使用建议还是在JDK8进行使用， 社区应该也在做相关的JDK升级的工作。

**Reply by alloyinfo** [reply]
换了java8也报这个错

**Reply by alloyinfo** [reply]
在broker_default.log中打印的这个错误，jps有namesrv和proxy
